### ISSUE TITLE
kubectl patch --type=strategic should prompt the name of the required key 

### ISSUE THREAD

**Issue by mfojtik** [kubernetes/kubernetes#57834]
When running `kubectl patch --type=strategic ...` (--type=strategic is by default)
```
$ kubectl patch svc multi-portsvc --type=strategic -p '
spec:
  ports:
  - targetPort: 441'
```

or 
```
$ oc patch rc rctest -p='
spec:
  template:
    spec:
      containers:
      - resources:
          limits:
            memory: 258Mi'
```

The output is:
```
The ReplicationController "rctest" is invalid: spec.template.spec.containers: Required value
```

Before, the output not ideal, but it at least showed the required key's name:
Error from server: map: map[targetPort:441] does not contain declared merge key: port

The recent version of kubectl seemed to dropped this message. 

Possibly related: https://github.com/kubernetes/kubernetes/pull/51321

**Comment by mfojtik** [comment]
@mengqiy FYI

**Comment by mengqiy** [comment]
> The ReplicationController "rctest" is invalid: spec.template.spec.containers: Required value

This error message comes from server-side validation.

SMP should return an error message when the merge key is missing.

Sent a fix #57854
